budgetApp1/budgetApp1/BudgetViewController.swift

```diff
@@ -9,49 +9,73 @@ import UIKit
+    private lazy var transactionViewModel = AccountTransactionsViewModel(repository: TransactionRepository(), delegate: self)
     
+    // MARK: - Functions
     override func viewDidLoad() {
         super.viewDidLoad()
-        
-        viewModel = BudgetViewModel(repository: BudgetRepository(), delegate: self)
-        viewModel?.getBudgets()
-        // Do any additional setup after loading the view.
-         
+        budgetViewModel.fetchBudgets()
+        transactionViewModel.fetchTransactions()
+        setupTableView()
     }
 
-    private func fillUI() {
-        if !isViewLoaded {
-            return
-        }
-        
-        guard let viewModel = viewModel, let budgetList = viewModel.budgetList, !budgetList.isEmpty else {
-            return
-        }
-        budgetNameLabel.text = budgetList.first?.name
-        startDateLabel.text = budgetList.first?.firstMonth
-        endDateLabel.text = budgetList.first?.lastMonth
+    private func updateView(withBudget budget: Budget) {
+        budgetNameLabel.text = budget.name
+        startDateLabel.text = budget.firstMonth
+        endDateLabel.text = budget.lastMonth
+        toDateLabel.text = "–"
+        recentTransactionsLabel.text = "Most recent transactions"
     }
+    
+    private func setupTableView() {
+        tableView.delegate = self
+        tableView.dataSource = self
+        tableView.register(TransactionTableViewCell.nib(), forCellReuseIdentifier: NibIdentifiers.TransactionViewCellIdentifier)
+        tableView.separatorStyle = .singleLine
+        tableView.separatorColor = uiSpecs.primaryColourOne
+        tableView.separatorInset = UIEdgeInsets(top: 0, left: 16, bottom: 0, right: 16)
+    }
+}
 
+extension BudgetViewController: UITableViewDelegate, UITableViewDataSource {
+    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        60.0
+    }
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        transactionViewModel.transactionListCount
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        guard let cell = tableView.dequeueReusableCell(withIdentifier: NibIdentifiers.TransactionViewCellIdentifier) as? TransactionTableViewCell else { return UITableViewCell() }
+        guard let transaction = transactionViewModel.transaction(atIndex: indexPath.item) else { return UITableViewCell() }
+        cell.populateWith(transaction: transaction)
+        return cell
+    }
 }
 
 // MARK: - ViewModel Delegate
-
-extension BudgetViewController: ViewModelDelegate {
+extension BudgetViewController: ViewModelDelegate, AccountsTransactionViewModelDelegate {
     
     func reloadView() {
-        fillUI()
+        guard let budget = budgetViewModel.budgetList?.first else { return }
+        updateView(withBudget: budget)
+        
```

Comment:
unnecessary new line

Reply:
I was using to show that one line is for the ViewModelDelegate and the other line for AccountsTransactionViewModelDelegate, however, I see that it is unnecessary now.